staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller.go

```diff
@@ -390,31 +390,42 @@ func (c *celAdmissionController) Validate(
 
 	if len(deniedDecisions) > 0 {
 		// TODO: refactor admission.NewForbidden so the name extraction is reusable but the code/reason is customizable
-		var message string
 		deniedDecision := deniedDecisions[0]
-		if deniedDecision.Binding != nil {
-			message = fmt.Sprintf("ValidatingAdmissionPolicy '%s' with binding '%s' denied request: %s", deniedDecision.Definition.Name, deniedDecision.Binding.Name, deniedDecision.Message)
-		} else {
-			message = fmt.Sprintf("ValidatingAdmissionPolicy '%s' denied request: %s", deniedDecision.Definition.Name, deniedDecision.Message)
+		var causes []metav1.StatusCause
+		for _, decision := range deniedDecisions {
+			var message string
+			if decision.Binding != nil {
+				message = fmt.Sprintf("ValidatingAdmissionPolicy '%s' with binding '%s' denied request: %s", decision.Definition.Name, decision.Binding.Name, decision.Message)
+			} else {
+				message = fmt.Sprintf("ValidatingAdmissionPolicy '%s' denied request: %s", decision.Definition.Name, decision.Message)
+			}
+
+			//!TODO: does it make sense to use decision.Reason for Cause.Type?
```

Comment:
I know we can't use it directly since it is a different type, just seems odd that every rule provides a `Reason` but we only use it first rule we happen to evaluate.

Reply:
I didn't expect code change in this PR.
We didn't provide the cause type since we didn't expect user to check the type based on what is inside details.Causes. I agree it would be a good candidate for further improvement.

Reply:
discussed offline that we will keep the bug fix but remove todo